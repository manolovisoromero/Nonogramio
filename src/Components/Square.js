import React, { Component } from 'react';


export default class Square extends Component {
    constructor(props) {
      super(props)
      this.state = {
        fieldcell: props.field,
        value: props.value,
        clas: props.clas,
      }
      this.handleClick = this.handleClick.bind(this);
  
  
    }
  
    componentDidMount() {
  
    }

 
    handleClick() {
      if(this.state.fieldcell){
        if (this.state.clas === 'clickedSquare') {
          this.setState({
            clas: 'square'
          })
        } else {
          this.setState({
            clas: 'clickedSquare'
          })
        }
      }
    }
    render() {
      return (
        <button className={this.state.clas} onClick={this.handleClick} background={this.state.backgroundcolor}>
          {this.state.value !== 0 ? this.state.value : null}
        </button>
      );
    }
  }