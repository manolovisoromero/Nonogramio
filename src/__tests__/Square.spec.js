import Square from '../Components/Square'
import React from 'react';
import { create } from "react-test-renderer";
import {mount} from 'enzyme';


describe("Sqaure Component", () => {
    test("Labelsquare render", () => {
      const button = create(<Square clas={'labelSquare'}  value={1} />);
      expect(button.toJSON()).toMatchSnapshot();
    });

    test("Field square render ", () => {
        const button = create(<Square clas={'square'}  field={true} />);
        expect(button.toJSON()).toMatchSnapshot();
      });
    
  });



