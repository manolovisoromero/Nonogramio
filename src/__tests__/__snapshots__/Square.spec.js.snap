// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sqaure Component Field square render  1`] = `
<button
  className="square"
  onClick={[Function]}
/>
`;

exports[`Sqaure Component Labelsquare render 1`] = `
<button
  className="labelSquare"
  onClick={[Function]}
>
  1
</button>
`;

exports[`Sqaure Component Matches the snapshot 1`] = `
<button
  className="labelSquare"
  onClick={[Function]}
>
  1
</button>
`;

exports[`Sqaure Component Matches the snapshot 2  1`] = `
<button
  className="row"
  onClick={[Function]}
/>
`;
